src/d3d9/d3d9_device.cpp

```diff
@@ -4681,6 +4723,7 @@ namespace dxvk {
     if (tex != nullptr && tex->IsRenderTarget())
       m_activeRTTextures |= bit;
 
+    UpdateManagedTextureUploadIndex(index);
```

Comment:
Most of the code in `UpdateManagedTextureUploadIndex` is the same as in this function (computin the index and fetching the texture pointer etc). Calling `UpdateManagedTextureUploadsUsed` for every single texture also seems unnecessary, so it would seem useful to rewrite this as
```c++
if (tex != nullptr) {
  if (unlikely(tex->IsRenderTarget())
    m_activeRTTextures |= bit;
  if (unlikely(tex->NeedsAnyUpload())
    m_activeTexturesToUpload |= bit;
}
```

Then maybe do `UpdateManagedTextureUploadsUsed` during the upload.